lib/ruby_indexer/lib/ruby_indexer/declaration_listener.rb

```diff
@@ -751,6 +753,41 @@ def handle_module_operation(node, operation)
       end
     end
 
+    sig { params(node: Prism::CallNode).void }
+    def handle_module_function(node)
+      arguments_node = node.arguments
+      return unless arguments_node
+
+      name_argument = arguments_node.arguments.first
+      return unless name_argument.is_a?(Prism::SymbolNode)
+
+      method_name = name_argument.value.to_s
+      owner_name = @owner_stack.last&.name
+      return unless owner_name
+
+      entries = @index.resolve_method(method_name, owner_name)
+      return unless entries
+
+      entries.each do |entry|
+        entry_owner_name = entry.owner&.name
+        next unless entry_owner_name
+
+        entry.visibility = Entry::Visibility::PRIVATE
+
+        singleton = @index.existing_or_new_singleton_class(entry_owner_name)
+        @index.add(Entry::Method.new(
```

Comment:
This `@index.add` should happen only once outside of this loop. The `resolve_method` method returns an array to account for methods that are overridden, but there's no need to create multiple entries for the singleton version of the method.

Let's move it after the loop. This loop should only set all entries' visibilities to private and then we add the singleton version to the index.

To get the singleton class, you can use the same `owner_name` you used to resolve the method:

```ruby
singleton = @index.existing_or_new_singleton_class(owner_name)
```

Reply:
Ignore this. See https://github.com/Shopify/ruby-lsp/pull/2733#discussion_r1808866464